javalin/src/main/java/io/javalin/plugin/rendering/vue/JavalinVue.kt

```diff
@@ -60,22 +65,23 @@ object JavalinVue {
         |        state: ${JavalinJson.toJson(state ?: stateFunction(ctx))}
         |    }""".trimMargin() + "\n</script>\n"
 
-    internal fun setRootDirPathIfUnset(ctx: Context) {
-        vueDirPath = vueDirPath ?: if (ctx.isLocalhost()) Paths.get("src/main/resources/vue") else PathMaster.classpathPath("/vue")
+    internal fun setRootDirPathIfUnset() {
+        vueDirPath = vueDirPath ?: if (isDev) Paths.get("src/main/resources/vue") else PathMaster.classpathPath("/vue")
     }
 
     private fun String.replaceWebjarsWithCdn() =
-            this.replace("@cdnWebjar/", if (useCdn) "https://cdn.jsdelivr.net/webjars/org.webjars.npm/" else "/webjars/")
+            this.replace("@cdnWebjar/", if (isDev) "/webjars/" else "https://cdn.jsdelivr.net/webjars/org.webjars.npm/")
 
 }
 
+
 class VueComponent @JvmOverloads constructor(private val component: String, private val state: Any? = null) : Handler {
     override fun handle(ctx: Context) {
-        JavalinVue.setRootDirPathIfUnset(ctx)
-        JavalinVue.useCdn = !ctx.isLocalhost()
+        JavalinVue.isDev = JavalinVue.isDevFunction(ctx)
+        JavalinVue.setRootDirPathIfUnset()
```

Comment:
Since you would typically have many `VueComponent`s, all calling this handle function, might it make sense to lift these two global operations out to another place? Kinda similar to the state param, which some people expect to be local; i.e. you could have a race-condition if two threads try to set `isDevFunction` at the same time.

Reply:
`JavalinVue.setRootDirPathIfUnset()` is a one-off. We could treat `isDev` the same: Make it a nullable boolean and set it based on the first request?

Reply:
Probably better. I think I would expect it to either be set once, globally, and/or pass it for each request, similar to state.

Reply:
Fixed in newest patchset!